compiler/rustc_hir_typeck/src/expr_use_visitor.rs

```diff
@@ -443,7 +443,7 @@ impl<'a, 'tcx> ExprUseVisitor<'a, 'tcx> {
                         if matches!((lhs, wild, rhs), (&[], Some(_), &[]))
                             // Arrays have a statically known size, so
                             // there is no need to read their length
-                            || discr_place.place.base_ty.is_array()
+                            || discr_place.place.base_ty.peel_refs().is_array()
```

Comment:
should we be doing `discr_place.place.ty()` insead of checking `base_ty`? Idk if that's right, will need checking...